fhir-package-source/package-source.rst

```diff
@@ -0,0 +1,83 @@
+	FhirPackageSource resolver = new(new string[] { package1, package 2});
+
+However, if you don't have the packages you want to resolve your artifacts from yet, you can specify a FHIR package server, and the names and versions (using an ``@``) of the packages to install the packages locally and create a ``FhirPackageSource`` immediately.
+
+.. code:: csharp
+	
+	var package1 = "hl7.fhir.r3.core.xml@3.0.2";
+	var package2 = "hl7.fhir.r3.expansions@3.0.2";
+	FhirPackageSource resolver = new("https://packages.simplifier.net", new string[] { package1, package 2});
+
+.. note:: You may need an active internet connection to use this. This function will first check if the specified packages are already installed locally before searching for the packages online.
+	
+
+How to resolve artifacts from packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The most common use case for the FHIR Package source is to resolve conformance resources like ``StructureDefinitions``, ``ValueSets``, and ``CodeSystems``  from specific FHIR packages. 
+
+All conformance resources can be resolved by specifying their canonical url.
+
+.. code:: csharp
+
+	FhirPackageSource resolver = new(new string[] { package1, package 2});
+	var patientProfile = await resolver.ResolveByCanonicalUriAsync("http://hl7.org/fhir/StructureDefinition/Patient") as StructureDefinition;
+	var genderValueSet = await resolver.ResolveByCanonicalUriAsync("http://hl7.org/fhir/ValueSet/administrative-gender") as ValueSet;
+
+You can also find any resource by their type and id, or any file by their name or relative filepath.
+
+.. code:: csharp
+
+	FhirPackageSource resolver = new(new string[] { package1, package 2});
+
+	//find resource by id
+	StructureDefinition pat = await resolver.ResolveByUriAsync("StructureDefinition/Patient") as StructureDefinition;
+
+	//find file by name
+	Stream file = resolver.LoadArtifactByName("StructureDefinition-Patient.xml");
+
+	//find file by path
+	Stream file2 = resolver.LoadArtifactByPath("package/StructureDefinition-Patient.xml");
+
+Next to that, there are some specific functions for certain resource types.
+
+.. code:: csharp
+
+	FhirPackageSource resolver = new(new string[] { package1, package 2});
+
+	//Find CodeSystems by ValueSets
+	CodeSystem cs = resolver.FindCodeSystemByValueSet("http://hl7.org/fhir/ValueSet/address-type");
+
+	//Find ConceptMaps by source and/or target (sourceUri or targetUri can be null)
+	ConceptMap cms = resolver.FindConceptMaps(sourceUri: "http://hl7.org/fhir/ValueSet/data-absent-reason", targetUri: "http://hl7.org/fhir/ValueSet/v3-NullFlavor");
+
+	//Find NamingSystem by uniqueId
+	NamingSystem ns = resolver.FindNamingSystem("http://snomed.info/sct");
+
+
+Different FHIR Versions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There is also a ``class`` in the SDK called ``CommonFhirPackageSource``. This ``class`` is located in the FHIR version agnostic part of the SDK, and this is actually where all the magic happens.
```

Comment:
You don't have to `class` the word class, I have not done so in the rest of the documentation.